### PR TITLE
769 migrate mutations action to new framework

### LINKED ISSUE #769: Migrate `Mutations` action to new framework
See #758

### PR DESCRIPTION
**Pull request by Taepper**
resolves #769

### Summary

This was a rather straight-forward implementation, moving the custom implementation to a `arrow::acero::SourceNode`.

## PR Checklist

- [X] All necessary documentation has been adapted or there is an issue to do so.
- [X] The implemented feature is covered by an appropriate test.

### CODE REVIEW

File: src/silo/query_engine/actions/mutations.cpp [hunk 1]
```diff
@@ -295,54 +307,115 @@ QueryResult Mutations<SymbolType>::execute(
...
+            ARROW_RETURN_NOT_OK(addMutationsToOutput(
+               sequence_name,
+               *sequence_column_metadata,
+               given_min_proportion,
+               bitmaps_to_evaluate.at(sequence_name),
+               output_builder
+            ));
+         }
+      }
+      // Order of result_columns is relevant
```
Comment (fengelniederhammer) [inline src/silo/query_engine/actions/mutations.cpp:356]: Maybe we should also say why it's relevant?

**Review by fengelniederhammer — APPROVED** [review]
Looks good :) 

What do you think about adding more unit tests? I didn't find a test that sets a `sequenceName`. There are probably some more edge cases that are not covered yet.

**Comment by Taepper** [comment]
@fengelniederhammer just pinging because I think you will like the second refactor commit :)
